sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/MessageToStringConverter.cs

```diff
@@ -20,52 +21,73 @@ public async Task<string> ConvertAsync(ServiceBusReceivedMessage input, Cancella
             {
                 throw new ArgumentNullException(nameof(input));
             }
-            if (input.Body == null)
-            {
-                return null;
-            }
 
-            Stream stream = input.Body.ToStream();
-
-            TextReader reader = new StreamReader(stream, StrictEncodings.Utf8);
-            try
+            var body = input.GetRawAmqpMessage().Body;
+            if (body.BodyType == AmqpMessageBodyType.Data)
             {
-                cancellationToken.ThrowIfCancellationRequested();
-                try
+                if (input.Body == null)
                 {
-                    return await reader.ReadToEndAsync().ConfigureAwait(false);
-                }
-                catch (DecoderFallbackException)
-                {
-                    // we'll try again below
+                    return null;
                 }
 
-                // We may get here if the message is a string yet was DataContract-serialized when created. We'll
-                // try to deserialize it here using GetBody<string>(). This may fail as well, in which case we'll
-                // provide a decent error.
+                Stream stream = input.Body.ToStream();
 
+                TextReader reader = new StreamReader(stream, StrictEncodings.Utf8);
                 try
                 {
-                    var serializer = DataContractBinarySerializer<string>.Instance;
-                    stream.Position = 0;
-                    return (string)serializer.ReadObject(stream);
+                    cancellationToken.ThrowIfCancellationRequested();
+                    try
+                    {
+                        return await reader.ReadToEndAsync().ConfigureAwait(false);
```

Comment:
I got all excited that `Encoding.GetString` has a `ReadOnlySpan<byte>` overload that we could use here... and then realized that it doesn't exist in `netstandard2.0`.   😦